core/domain/user_services.py

```diff
@@ -551,3 +556,16 @@ def get_user_impact_score(user_id):
         return model.impact_score
     else:
         return 0
+
+def fetch_gravatar(email):
+    """Returns the gravatar corresponding to given email"""
```

Comment:
Some test specs:
- fetch_gravatar('...') makes exactly one URLfetch request (you can use the CallCounter in test_utils to track this)
- if there's a bad connection and the fetch fails, the return value here should equal DEFAULT_IDENTICON_DATA_URL and log an error (so, you may need to swap out logging.error as well, here)